server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/reader/CobolLineReaderImpl.java

```diff
@@ -95,33 +99,23 @@
     CobolLine cobolLine = new CobolLine();
     line = getDelegate().apply(line);
 
-    ResultWithErrors<String> result = checkLineLength(line, uri, lineNumber);
-    String adjustedLine = result.getResult();
-    List<SyntaxError> errors = new ArrayList<>(result.getErrors());
-
-    if (adjustedLine.length() > 0) {
-      for (int i = adjustedLine.length(); i > 0; i--) {
-        if (i > 72) {
-          cobolLine.setCommentArea(adjustedLine.substring(72, i));
-          i = 73;
-        } else if (i > 11) {
-          cobolLine.setContentAreaB(adjustedLine.substring(11, i));
-          i = 12;
-        } else if (i > 7) {
-          cobolLine.setContentAreaA(adjustedLine.substring(7, i));
-          i = 8;
-        } else if (i > 6) {
-          String indicatorArea = adjustedLine.substring(6, 7);
-          ResultWithErrors<CobolLineTypeEnum> type = determineType(indicatorArea, uri, lineNumber);
-          cobolLine.setIndicatorArea(indicatorArea);
-          cobolLine.setType(type.getResult());
-          errors.addAll(type.getErrors());
-          i = 7;
-        } else {
-          cobolLine.setSequenceArea(adjustedLine.substring(0, i));
-          i = 0;
-        }
-      }
+    List<SyntaxError> errors = new ArrayList<>();
+
+    Matcher matcher = COBOL_LINE_PATTERN.matcher(line);
+    matcher.matches();
```

Comment:
I expect we will have a code smell about ignored return value here.

Reply:
In theory, any line must match this pattern. I'll log an error in case the line didn't match.